typescript/sdk/src/token/EvmERC20WarpModule.ts

```diff
@@ -52,15 +64,88 @@ export class EvmERC20WarpModule extends HyperlaneModule<
   /**
    * Updates the Warp Route contract with the provided configuration.
    *
-   * @remark Currently only supports updating ISM or hook.
-   *
    * @param expectedConfig - The configuration for the token router to be updated.
    * @returns An array of Ethereum transactions that were executed to update the contract, or an error if the update failed.
    */
   public async update(
-    _expectedConfig: TokenRouterConfig,
+    expectedConfig: TokenRouterConfig,
   ): Promise<AnnotatedEV5Transaction[]> {
-    throw Error('Not implemented');
+    TokenRouterConfigSchema.parse(expectedConfig);
+    const actualConfig = await this.read();
+
+    const updateTransactions: AnnotatedEV5Transaction[] = [];
+    const updateIsmTx = await this.updateIsm(actualConfig, expectedConfig);
+    if (updateIsmTx) {
+      updateTransactions.push(updateIsmTx);
+    }
+
+    return updateTransactions;
+  }
```

Comment:
a lot of the read and checking of ism diffs is already built into the ISM update() function, and because part of the route config is the ISM, I would've expected this module to have some sort of `warpIsm` as part of the class

this would let you condense the update into one function that better leverages the update logic of the ISM module
```ts
  /**
   * Updates the Warp Route contract with the provided configuration.
   *
   * @param expectedConfig - The configuration for the token router to be updated.
   * @returns An array of Ethereum transactions that were executed to update the contract, or an error if the update failed.
   */
  public async update(
    expectedConfig: TokenRouterConfig,
  ): Promise<AnnotatedEV5Transaction[]> {
    expectedConfig = TokenRouterConfigSchema.parse(expectedConfig);

    if (!expectedConfig.interchainSecurityModule) {
      // If the ISM is not configured, we can't update the ISM
      // TODO: log stuff?
      return [];
    }

    const updateTxs = await this.warpIsm.update(
      expectedConfig.interchainSecurityModule,
    );

    const contractToUpdate = MailboxClient__factory.connect(
      this.args.addresses.deployedTokenRoute,
      this.multiProvider.getProvider(this.args.chain),
    );

    const currentIsmAddress = await contractToUpdate.interchainSecurityModule();
    const newIsmAddress = this.warpIsm.serialize().deployedIsm;
    if (!eqAddress(currentIsmAddress, this.warpIsm.serialize().deployedIsm)) {
      updateTxs.push({
        annotation: `Setting ISM for Warp Route to ${newIsmAddress}`,
        chainId: Number(this.multiProvider.getChainId(this.args.chain)),
        to: contractToUpdate.address,
        data: contractToUpdate.interface.encodeFunctionData(
          'setInterchainSecurityModule',
          [newIsmAddress],
        ),
      });
    }

    return updateTxs;
  }
```